### DISCUSSION TITLE
Neuralnet tracker: can't open camera

### DISCUSSION THREAD

**Discussion by theluckyone17** [opentrack/opentrack#1864] [General]
Microsoft pushed their Win10 EOL announcement to my gaming box earlier this week, which drove me to start moving my Star Citizen gaming back over to my Gentoo box. I've got nearly everything up and running, with the exception of my TrackHat v2 sensor face tracking. I've managed to compile OpenTrack from source; took a bit to get there, including the NeuralNet tracker input module. However, it's throwing an error claiming that the tracker cannot open the camera. This is despite my being able to preview the camera output within the Tracker->Camera Settings function. I've tested it against another webcam (Logitech C930e, off the top of my head), with the same results. 

The console drops the following error:
`DEBUG [/home/noel/src/opentrack/tracker-neuralnet/ftnoir_tracker_neuralnet.cpp:542]: neuralnet tracker: can't open camera`

Any thoughts?

**Comment by theluckyone17** [comment]
I see 2024.1.1 has gone live. I'll give that a try. If that doesn't work, and assuming this isn't the proper channel, should I be looking elsewhere?

**Comment by sthalik** [comment]
Upgrading opentrack won't do anything here.

See why `/dev/video0` and similar device nodes aren't available. Most likely the camera isn't detected at all, or you're not in the group that allows for using it.

**Comment by sthalik** [comment]
To use tracker -> camera settings you need to install `qv4l2`.

**Comment by theluckyone17** [comment]
Greatly appreciate the replies! Didn't think updating was going to be the resolution, but I'll be heading down that road eventually anyway.

qv4l2 is installed, and I'm able to access the /dev/video0 device node. I'm able to launch qv4l2 and preview the streaming video without issue:
![qv4l2](https://github.com/opentrack/opentrack/assets/9260007/6e959c6a-9171-454a-be51-b8ca1c90a7b1)

I'm then able to:
- exit qv4l2 gracefully
- launch OpenTrack
- click the **Options** button
- click the **Tracker** tab
- click the **Camera settings** button
- click the Start Capturing button
- and preview the streaming video without issue: 
![opentrack2](https://github.com/opentrack/opentrack/assets/9260007/c03c8092-b8cf-421e-8b0a-46acc8b1a558)
- close the **capture**, **v4l2 test branch**, and **Options** windows
- click the **Start** button
- tracking fails to start.
- the following text is dumped into the terminal (username obfuscated):
`DEBUG [/home/USER/src/opentrack/proto-wine/ftnoir_protocol_wine.cpp:81]: proto/wine: wine_path: "/home/USER/.local/share/lutris/runners/wine/lutris-GE-Proton8-10-x86_64/bin/wine"
DEBUG [/home/USER/src/opentrack/proto-wine/ftnoir_protocol_wine.cpp:111]: proto/wine: wineprefix: "/home/USER/Games/star-citizen"
DEBUG [/home/USER/src/opentrack/proto-wine/ftnoir_protocol_wine.cpp:137]: proto/wine: shm success
[ INFO:0@1.608] global registry_parallel.impl.hpp:96 ParallelBackendRegistry core(parallel): Enabled backends(3, sorted by priority): ONETBB(1000); TBB(990); OPENMP(980)
[ INFO:1@1.719] global videoio_registry.cpp:244 VideoBackendRegistry VIDEOIO: Enabled backends(7, sorted by priority): FFMPEG(1000); GSTREAMER(990); INTEL_MFX(980); CV_IMAGES(970); CV_MJPEG(960); UEYE(950); OBSENSOR(940)
DEBUG [/home/USER/src/opentrack/tracker-neuralnet/ftnoir_tracker_neuralnet.cpp:542]: neuralnet tracker: can't open camera
`

There's obviously something that the neuralnet tracker doesn't like about that device, so:
- close OpenTrack
- physically disconnect the Trackhat Sensorv2 cam
- physically connect the other web cam I have available (might not be a Logitech; it's reading as a HD Web Cam, VID:05a3, PID:9331)
- repeat the steps above, previewing the video output first, then starting the tracker
- and receive the same result and text output at the terminal.

**Reply by sthalik** [reply]
Seems like `opencv-videoio` is compiled without V4L support.

**Reply by theluckyone17** [reply]
Good call! I missed that use flag; Media-libs/opencv was built without v4l support. Rebuilding now.

Update: That did it, it's working perfectly now. Thanks for the help!
